connmgr/tor.go

```diff
@@ -106,24 +106,39 @@ func TorLookupIP(host, proxy string) ([]net.IP, error) {
 		}
 		return nil, ErrTorInvalidProxyResponse
 	}
-	if buf[3] != 1 {
-		err := torStatusErrors[torGeneralError]
-		return nil, err
-	}
 
-	buf = make([]byte, 4)
-	bytes, err := conn.Read(buf)
-	if err != nil {
-		return nil, err
-	}
-	if bytes != 4 {
-		return nil, ErrTorInvalidAddressResponse
-	}
+	addr := make([]net.IP, 1)
 
-	r := binary.BigEndian.Uint32(buf)
+	switch buf[3] {
```

Comment:
Can we extract some test cases from either the Tor docs, or that Python tool so we can verify the response parsing?

Reply:
This isn't tor-specific -- this is the socks5 response format. We might be able to use a socks5 library, but I don't know if the library could let us specify the `0xF0` byte for "Tor resolve"